ddl/placement_policy.go

```diff
@@ -269,6 +294,36 @@ func checkPlacementPolicyNotInUseFromInfoSchema(is infoschema.InfoSchema, policy
 	return nil
 }
 
+func getPlacementPolicyDependedObjectsIDs(t *meta.Meta, policy *model.PolicyInfo) ([]int64, error) {
+	schemas, err := t.ListDatabases()
+	if err != nil {
+		return nil, err
+	}
+	ids := make([]int64, 0, len(schemas))
+	for _, dbInfo := range schemas {
+		if dbInfo.PlacementPolicyRef != nil && dbInfo.PlacementPolicyRef.ID == policy.ID {
+			ids = append(ids, dbInfo.ID)
+		}
+		tables, err := t.ListTables(dbInfo.ID)
+		if err != nil {
+			return nil, err
+		}
+		for _, tblInfo := range tables {
+			if ref := tblInfo.PlacementPolicyRef; ref != nil && ref.ID == policy.ID {
+				ids = append(ids, tblInfo.ID)
+			}
+			if tblInfo.Partition != nil {
+				for _, part := range tblInfo.Partition.Definitions {
+					if part.PlacementPolicyRef != nil && part.PlacementPolicyRef.ID == part.ID {
```

Comment:
If `part. PlacementPolicyRef == nil` but `tblInfo.PlacementPolicyRef.ID == policy.ID` partition should also depend on the policy.

In the design doc: https://github.com/pingcap/tidb/blob/master/docs/design/2020-06-24-placement-rules-in-sql.md#partitioned-tables

> The semantics of partitioning are different to the default database policy. When a table is partitioned, the partitions will by default inherit the same placement policy as the table.

Reply:
ye, the inherit work haven't been done yet.
(table inherit db, partition inherit table, and some other details should be dig out. I will done it in next PR)